Add forge ingot, dust, nugget and storage block tags

Lemon-Juiced opened the pull request:
Firstly, I cannot figure out how to decouple the SAG Mill recipes from the push as I added them over a week ago and can't figure out how to only push the tags. This push was originally only meant to be the tags. 

I have implemented tags for EIO's alloys, both the ingot and nugget forms. This is very good for cross-mod compatibility as it allows other mods, primarily add-ons, to use EIO's resources from a tag in the forge directory as opposed to directly declaring the item. This would also allow mods like Thermal Expansion to have recipes for EIO alloys in its Induction Smelter without having missing tags in JEI for recipes when EIO is not installed.

I'd personally also like to add one for Grains of Infinity (for mods like Mystical Agriculture) however, I do not know how that would go with the style guidelines of the mod and if the grains of infinity are considered a dust or closer to silicon where they have their own tag.

For the SAG Mill recipes, I replicated what was created for ores in 1.12.2 on the raw resources, to be balanced with other current mods like Thermal Expansion and Mekanism. This is technically still under RFC, so I understand it not being implemented, refer to the first paragraph. However, if you do decide to implement it they will be usable in survival until a permanent answer is implemented; we have to be careful if we go this route as players might try to do something auto-crafting wise and this could break that.

This shouldn't break anything as it is just datagen.

- [X] My code follows the style guidelines of this project (.editorconfig, most IDEs will use this for you).
- [X] I have performed a self-review of my own code.
- [X] I have commented my code in areas it may be challenging to understand. <!-- (Although we prefer code that is readable instead of over-commented) -->
- [X] I have made corresponding changes to the documentation.
- [x] My changes are ready for review from a contributor.

Lemon-Juiced committed:
Raw Ore Outputs in Sag Mill, corresponding to issue 327 in main repo.

Lemon-Juiced committed:
Ingot and Nugget tags for better cross-mod compat.

Lemon-Juiced committed:
Merge branch 'Team-EnderIO:dev/1.20.1' into dev/1.20.1

Rover656 commented:
I am quite happy with the approach to adding the tags for ingots and nuggets, and I would suggest the addition of the block tag `storage_blocks` too.
As for the ore conversions, I think that these are fairly sensible recipes, however I'd like to separate these two "features" out separately, as tags are completely fine and basically already ready to go in, but the raw ores might need some back and forth.

So my suggestion would be to move the raw ores to another PR for further discussion as I think you're nearly there, then add the `storage_blocks` tag for the alloy blocks, then this can be merged in.

Thanks a lot for contributing!

Rover656 commented:
Also to address grains of infinity, I think it'd be reasonable to have dusts/infinity (or grains_of_infinity if infinity is too general)

Rover656 reviewed (changes requested):
Requested changes as listed in github discussion (github made me add a message :P)

Lemon-Juiced commented:
I completely understand the ores need back and forth, I couldn't figure out how to only give one commit instead of both. I'll look into adding the other tags tonight. I'm just a bit busy at the moment (I'll also see if I can remove the ore conversions from the request).

You're very welcome for the contribution, Ender IO has always been a favorite of mine and I'm happy to see it's coming back.

Lemon-Juiced committed:
Merge branch 'Team-EnderIO:dev/1.20.1' into dev/1.20.1

Lemon-Juiced committed:
Made requested changes for #424

Lemon-Juiced commented:
These changes should have addressed the issues, outlined above. I just removed the SAG mill recipes from the git instead of trying to save the code separately. I also went ahead and added dust tags for the other grains while I was at it.

Rover656 reviewed (dismissed):
This looks good to me, thanks a lot!!

Rover656 committed:
Fixed storage_block item tag generation and formatting
